### ISSUE TITLE
[ENH] Fix failing output tests for the GreedyGaussianSegmentation detector

### ISSUE THREAD

**Issue by Alex-JG3** [sktime/sktime#7412]
## Summary

The following tests for the GreedyGaussianSegmentation detector fail and thus have been added to the list of excluded tests:

* `test_predict_points`
* `test_predict_segments`
* `test_transform_output_type`
* `test_inheritance`
* `test_create_test_instance`

We should fix GreedyGaussianSegmentation to make sure these tests pass.

## Related PR

The tests were added in #6958.

**Comment by Spinachboul** [comment]
Would love working on this issue
@Alex-JG3 
Could I be assigned this one
went through the context of the previous issues and feel like some work needs to be done in the `GreedyGaussianSegmentation` class!
Would soon come with the snippets of the excluded tests passing soon.

**Comment by Alex-JG3** [comment]
Yes of course. I have assigned you now.

**Comment by Spinachboul** [comment]
@Alex-JG3 
Thanks!!

**Comment by Spinachboul** [comment]
@Alex-JG3 
Do we have the logs of the failing tests, then would be easier to refactor
Currently I have changed the implementation styles of several function in the GreedyGaussianSegmentation class, which might be the results of failing tests

Here is a brief summary for the same:

1. **Refactored Class Initialization:**
   - Simplified constructor by removing unnecessary arguments and ensuring compatibility with `GGS` class.

2. **Updated `fit_predict` Method:**
   - Merged `fit` and `predict` functionality into a single call, ensuring compatibility with `sklearn` interface.

3. **Enhanced Documentation:**
   - Added concise docstrings for methods, improved clarity on inputs and outputs.

4. **Optimized `fit` and `predict` Methods:**
   - Simplified input handling for time series data (support for `pd.Series`, `pd.DataFrame`, and `np.ndarray`).
   - Consolidated segmentation logic in `predict` to directly assign segment labels.

5. **Fixed Compatibility with Scikit-Learn Estimator Interface:**
   - Ensured `fit` method does not perform unnecessary operations, and refactored to maintain interface consistency.

6. **Updated Return Types:**
   - Ensured all methods return `np.ndarray` for segment labels as required by the interface.

**Comment by Alex-JG3** [comment]
These all sounds like good changes. Unfortunately I don't have logs of the failing tests. Are you able to produce the failing tests locally?

Here is an example script to get the `test_predict_points` to fail.

```python
# test.py

**Comment by Spinachboul** [comment]
@Alex-JG3 
I have did some more refactoring and I guess the tests are passing for `GreedyGaussianSegmentation` 
Here is the snippet for the same:

![image](https://github.com/user-attachments/assets/c5033321-2f40-41aa-b844-759a042ea17b)

Is this what is expected of the behaviour of the class?

**Comment by Alex-JG3** [comment]
That looks good. You can also pass a list of strings to `tests_to_run`, so you should be able to give it a list of all the failing tests.

Also, please consider opening a pull request with your changes. We'll then be able test your changes against the entire test suite.

**Comment by Alex-JG3** [comment]
Yes this looks good. Thanks for opening a PR.

**Comment by fkiraly** [comment]
Fixed in https://github.com/sktime/sktime/pull/7472
